docs/source/notebooks/wps_generate_climos.ipynb

```diff
@@ -0,0 +1,388 @@
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "<ipython-input-9-e3c8cb77b746>:9: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.\n",
+      "  for key in input_data.__dict__:\n",
+      "<ipython-input-9-e3c8cb77b746>:12: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.\n",
+      "  assert input_data.__dict__[key] == output_data.__dict__[key]\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Test normal output data is the same as the input data\n",
+    "input_data = Dataset(seasonal_opendap)\n",
+    "url = soup.get_text()\n",
+    "\n",
+    "with NamedTemporaryFile(suffix=\".nc\", prefix=\"tmp_copy\", dir=\"/tmp\", delete=True) as tmp_file:\n",
+    "    urlretrieve(url, tmp_file.name)\n",
+    "    output_data = Dataset(tmp_file.name)\n",
+    "\n",
+    "for key in input_data.__dict__:\n",
+    "    # History and frequency are different\n",
+    "    if key != 'frequency' and key != 'history':\n",
+    "        assert input_data.__dict__[key] == output_data.__dict__[key]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# process dry run for multiple files\n",
+    "dry_output = thunderbird.generate_climos(\n",
+    "    netcdf=[seasonal_opendap, annual_local], \n",
+    "    operation=operation, \n",
+    "    climo=climo, \n",
+    "    resolutions=resolutions, \n",
+    "    dry_run=dry_run\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[<metaurl mediatype=\"text/plain\">\n",
+      "     http://localhost:5001/outputs/5690ae31-0368-11eb-b7e1-c9fa1b0c3983/fdd_seasonal_CanESM2_rcp85_r1i1p1_1951-2100_dry.txt\n",
+      "    </metaurl>, <metaurl mediatype=\"text/plain\">\n",
+      "     http://localhost:5001/outputs/5690ae32-0368-11eb-b7e1-c9fa1b0c3983/gdd_annual_CanESM2_rcp85_r1i1p1_1951-2100_dry.txt\n",
+      "    </metaurl>]\n"
+     ]
+    }
+   ],
+   "source": [
+    "req = requests.get(dry_output.get()[0])\n",
+    "soup = BeautifulSoup(BeautifulSoup(req.content.decode('utf-8')).prettify()).find_all('metaurl')\n",
+    "print(soup)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Test dry output for multiple files\n",
+    "assert len(soup) == 2\n",
+    "    \n",
+    "url1 = soup[0].get_text()\n",
+    "with urlopen(url1) as f:\n",
+    "    output_data1 = f.read().decode('utf-8')\n",
+    "    \n",
+    "assert re.search(r'time_resolution: seasonal', output_data1)\n",
+    "    \n",
+    "url2 = soup[1].get_text()\n",
+    "with urlopen(url2) as f:\n",
+    "    output_data2 = f.read().decode('utf-8')\n",
+    "    \n",
+    "assert re.search(r'time_resolution: yearly', output_data2)"
```

Comment:
This entire cell is great! The tests are similar to what I suggested above where we can just look for items that interest us rather than matching everything word for word. It is clean, clear and simple :+1: 

There are some items we could clean up. For one, the same code snippet is repeated twice:
```
url1 = soup[0].get_text()
with urlopen(url1) as f:
    output_data1 = f.read().decode('utf-8')
    
assert re.search(r'time_resolution: seasonal', output_data1)
```
We can iterate over `soup` and squish the code under a loop (unintentional rhyme...):
```
for noodles, tr in zip(soup, ['seasonal', 'yearly']):
    url = noodles.get_text()
    with urlopen(url) as f:
        output_data = f.read().decode('utf-8')
    
    assert re.search(r'time_resolution: {}'.format(tr), output_data)
```